diff --git a/package/lean/luci-app-cpufreq/Makefile b/package/lean/luci-app-cpufreq/Makefile
index 758d279e0..f7a144056 100644
--- a/package/lean/luci-app-cpufreq/Makefile
+++ b/package/lean/luci-app-cpufreq/Makefile
@@ -7,7 +7,7 @@
 include $(TOPDIR)/rules.mk
 
 LUCI_TITLE:=LuCI for CPU Freq Setting
-LUCI_DEPENDS:=@arm
+LUCI_DEPENDS:=@(arm||aarch64)
 PKG_NAME:=luci-app-cpufreq
 PKG_VERSION:=1
 PKG_RELEASE:=6
