diff --git a/package/lean/luci-app-cpufreq/root/etc/uci-defaults/luci-cpufreq b/package/lean/luci-app-cpufreq/root/etc/uci-defaults/luci-cpufreq
index 8443be579..ce2b04399 100755
--- a/package/lean/luci-app-cpufreq/root/etc/uci-defaults/luci-cpufreq
+++ b/package/lean/luci-app-cpufreq/root/etc/uci-defaults/luci-cpufreq
@@ -7,5 +7,11 @@ uci -q batch <<-EOF >/dev/null
 	commit ucitrack
 EOF
 
+# Set default settings
+uci set cpufreq.@settings[-1].governor=schedutil
+uci set cpufreq.@settings[-1].minifreq=$(cat /sys/devices/system/cpu/cpufreq/policy0/cpuinfo_min_freq)
+uci set cpufreq.@settings[-1].maxfreq=$(cat /sys/devices/system/cpu/cpufreq/policy0/cpuinfo_max_freq)
+uci commit cpufreq
+
 rm -f /tmp/luci-indexcache
 exit 0
