diff --git a/feeds/luci/applications/luci-app-ipsec-vpnd/root/etc/uci-defaults/luci-ipsec b/feeds/luci/applications/luci-app-ipsec-vpnd/root/etc/uci-defaults/luci-ipsec
index 83b4c23..9214ace 100755
--- a/feeds/luci/applications/luci-app-ipsec-vpnd/root/etc/uci-defaults/luci-ipsec
+++ b/feeds/luci/applications/luci-app-ipsec-vpnd/root/etc/uci-defaults/luci-ipsec
@@ -10,12 +10,12 @@ uci -q batch <<-EOF >/dev/null
 EOF
 
 uci -q batch <<-EOF >/dev/null
-	delete network.VPN
-	set network.VPN=interface
-	set network.VPN.ifname="ipsec0"
-	set network.VPN.proto="static"
-	set network.VPN.ipaddr="10.10.10.1"
-	set network.VPN.netmask="255.255.255.0"
+	delete network.vpn
+	set network.vpn=interface
+	set network.vpn.ifname="ipsec0"
+	set network.vpn.proto="static"
+	set network.vpn.ipaddr="10.10.10.1"
+	set network.vpn.netmask="255.255.255.0"
 	
 	commit network
 		
@@ -53,19 +53,19 @@ uci -q batch <<-EOF >/dev/null
 	set firewall.@rule[-1].src="wan"
 	set firewall.@rule[-1].proto="esp"
   
-	delete firewall.VPN
-	set firewall.VPN=zone
-	set firewall.VPN.name="VPN"
-	set firewall.VPN.input="ACCEPT"
-	set firewall.VPN.forward="ACCEPT"
-	set firewall.VPN.output="ACCEPT"
-	set firewall.VPN.network="VPN"
-	
 	delete firewall.vpn
-	set firewall.vpn=forwarding
+	set firewall.vpn=zone
 	set firewall.vpn.name="vpn"
-	set firewall.vpn.dest="wan"
-	set firewall.vpn.src="VPN"
+	set firewall.vpn.input="ACCEPT"
+	set firewall.vpn.forward="ACCEPT"
+	set firewall.vpn.output="ACCEPT"
+	set firewall.vpn.network="vpn"
+
+	delete firewall.vpnf
+	set firewall.vpnf=forwarding
+	set firewall.vpnf.name="vpnf"
+	set firewall.vpnf.dest="wan"
+	set firewall.vpnf.src="vpn"
   
 	commit firewall
 EOF
